src/test/java/com/google/cloud/tools/maven/AbstractDeployMojoTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2018 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.maven;
+
+import java.lang.reflect.Field;
+import org.apache.maven.plugin.logging.Log;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AbstractDeployMojoTest {
+
+  private AbstractDeployMojo testMojo =
+      new AbstractDeployMojo() {
+        @Override
+        public void execute() {
+          // do nothing;
+        }
+      };
+
+  @Mock private Log mockLog;
+  private final Field projectField;
+
+  public AbstractDeployMojoTest() throws NoSuchFieldException {
```

Comment:
Oh I didn't actually know constructors could work on test classes like this - been always putting initialization code in `setUp`.

Reply:
It might be safer to write it in the setUp since it's technically mutable and shared in tests.